hibernate-core/src/main/java/org/hibernate/query/criteria/internal/predicate/InPredicate.java

```diff
@@ -197,14 +201,51 @@ public String render(boolean isNegated, RenderingContext renderingContext) {
 				}
 			}
 			else {
+				Expression lastValue = null;
 				buffer.append( '(' );
 				String sep = "";
 				for ( Expression value : values) {
+					lastValue = value;
 					buffer.append( sep )
 							.append( ( (Renderable) value )
 									.render( renderingContext ) );
 					sep = ", ";
 				}
+
+
+				if (renderingContext.getCriteriaLiteralHandlingMode() == LiteralHandlingMode.BIND &&
+						lastValue instanceof LiteralExpression) {
+
+					int bindValueCount = values.size();
+					int bindValueMaxCount = bindValueCount;
+
+					final SessionFactoryImplementor sfi = criteriaBuilder().getEntityManagerFactory().unwrap( SessionFactoryImplementor.class );
+					final Dialect dialect = sfi.getServiceRegistry().getService(JdbcServices.class).getJdbcEnvironment().getDialect();
+					final int inExprLimit = dialect.getInExpressionCountLimit();
+
+					boolean inClauseParameterPaddingEnabled = sfi.getSessionFactoryOptions().inClauseParameterPaddingEnabled() &&
+									bindValueCount > 2;
```

Comment:
swapping the parameter order around `&&` might be an idea for `bindValueCount > 2` is relatively lightweight and we can avoid some unnecessary cpu cycles due to short circuiting.

Reply:
It is just a copy&paste from QueryParameterBindingsImpl. I thought it is a goog idee to alter the code as little possible.